crates/agent/src/publications/specs.rs

```diff
@@ -300,6 +300,41 @@ async fn update_live_specs(
     Ok(lock_failures)
 }
 
+pub async fn check_source_capture_annotations(
+    draft: &tables::DraftCatalog,
+    pool: &sqlx::PgPool,
+) -> anyhow::Result<tables::Errors> {
+    let mut by_image: BTreeMap<String, bool> = BTreeMap::new();
+    let mut errors = tables::Errors::default();
+
+    for materialization in draft.materializations.iter() {
+        let Some(model) = materialization.model() else { return Ok(errors) };
+        let Some(image) = model.connector_image() else { return Ok(errors) };
+        let (image_name, image_tag) = split_image_tag(image);
+        let Some(connector_spec) = agent_sql::connector_tags::fetch_connector_spec(&image_name, &image_tag, pool).await? else { return Ok(errors) };
+        let resource_config_schema = connector_spec.resource_config_schema;
+
+        let resource_spec_pointers = crate::resource_configs::pointer_for_schema(resource_config_schema.0.get())?;
```

Comment:
Thinking through what would happen if a resource config schema was malformed or missing `x-collection-name`, and I don't see anywhere else where we really validate that. I'm realizing that we're currently missing an opportunity to validate the resource config schema in the `connector_tags` handler, which would allow us to fail fast in that case. Would you mind adding a check to `connector_tags.rs`?

Reply:
Good idea, added it and tested it manually by creating a local image of `materialize-sqlite` without the annotation to see the agent fail the connector_tags job:
<img width="1142" alt="image" src="https://github.com/user-attachments/assets/c791847a-a58f-4d46-b19d-a68a8559a9ea">